hw/ip/pwrmgr/doc/_index.md

```diff
@@ -0,0 +1,233 @@
+
+Ultimately when control is returned to software, it sees two reset reasons and must handle accordingly.
+
+
+## Wakeup Recording
+
+Similar to [reset handling](#Reset-Request-Handling), wakeup signals are only observed during slow FSM low power; however its recording is continuous until explicitly disabled by software.
+
+The wakeup recording begins when the fast FSM transitions out of ACTIVE state and continues until explicitly disabled by software.
+This ensures wakeup recording is continuous until software has set up the appropriate peripherals to handle future events.
+
+The software is also able to enable recording during ACTIVE state if it so chooses.  The recording enables are just OR’d together for hardware purposes.
+
+
+## Flash Handling
+
+### Power-Up Handling
+
+The AST automatically takes flash out of power down state as part of power manager's power up request.
+Once flash is powered up, the power manager keeps a running count of the flash initialization time.
+When this time has expired, an acknowledgement is forwarded to the flash controller.
+Once the boot ROM is allowed to execute, it is expected to initialize the flash prior to using it.
+This involves 4 steps:
+
+*   Poll flash control register to ensure flash has powered up (this is the same signal mentioned above).
+*   Obtain flash redundancy or configuration information (could be in flash or OTP)
+    *   If in flash, this information is unscrambled
+    *   This item is TBD and must be further discussed.
+*   Program redundancy or configuration information into the flash controller and lock it from further modification
+*   Initialize flash scrambling and programming any excluded regions based on OTP or other information.
+
+There may be additional steps depending on flash design and software needs.
+
+### Power-Down Handling
+
+When the device enters deep sleep, the flash is automatically put into power down mode by the AST.
+When the device exits low power state, the flash is automatically powered up and it is boot ROM responsibility to poll for flash power-up complete.
+
+
+### Flash Brownout Handling
+
+When the supply of the device dips below a certain point during a stateful flash operation (program or erase) the flash requires the operation to terminate in a pre-defined manner.
+This sequence will be exclusively handled by the AST.
+The power manager is unaware of the difference between POR and flash brownout.
+Because of this, the software also cannot distinguish between these two reset causes.
+
+
+## Supported Low Power Modes
+
+This section details the various low power modes supported by OpenTitan.
+
+
+### Deep Sleep or Standby
+
+This represents the lowest power mode of the device (outside of full power down or device held in reset).
+During this state, the following states of the device shall be true
+
+*   All clocks other than the KHz slow clock shall be turned off at the source.
+*   All non-always-on digital domains are powered off
```

Comment:
One of these lines reads like an RFC ("the client shall do XYZ") and the other sounds more informative. I don't think the distinction is intentional? I'd probably change to the second phrasing, so "clocks ... are turned off", unless there's a good reason not to.

Reply:
thanks for catching this, fixed!